app/models/assessment.rb

```diff
@@ -30,23 +28,16 @@ class Assessment < ApplicationRecord
   has_many :proceeding_types,
            dependent: :destroy
 
-  # Always instantiate a new Remarks object from a nil value
-  def remarks
-    attributes["remarks"] || Remarks.new(id)
-  rescue StandardError
-    Remarks.new(id)
-  end
-
   def proceeding_type_codes
     proceeding_types.order(:ccms_code).map(&:ccms_code)
   end
 
-  def add_remarks!(new_remarks)
-    my_remarks = remarks
-
-    new_remarks.each do |remark|
-      my_remarks.add(remark.type, remark.issue, remark.ids)
+  def transform_remarks(remarks)
+    remarks_hash_by_type = remarks.group_by(&:type)
+    remarks_hash = remarks_hash_by_type.transform_values do |v|
+      v.group_by(&:issue).transform_values { |c| c.map(&:ids).flatten }
     end
-    update!(remarks: my_remarks)
+    remarks_hash.merge! explicit_remarks.by_category
+    remarks_hash.symbolize_keys
   end
```

Comment:
This method looks like view code? Not sure it belongs on 'assessment'?

Reply:
@starswan Agree and Moved to RemarksDecorator